akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -292,27 +297,32 @@ private[akka] class Shard(
     log.error("Shard type [{}] id [{}] lease lost. Reason: {}", typeName, shardId, msg.reason)
     // Stop entities ASAP rather than send termination message
     context.stop(self)
-
   }
-  def receiveShardCommand(msg: ShardCommand): Unit = msg match {
+
+  private def receiveShardCommand(msg: ShardCommand): Unit = msg match {
+    // those are only used with remembering entities
     case RestartEntity(id)    => getOrCreateEntity(id)
     case RestartEntities(ids) => restartEntities(ids)
   }
 
-  def receiveStartEntity(start: ShardRegion.StartEntity): Unit = {
+  private def receiveStartEntity(start: ShardRegion.StartEntity): Unit = {
     val requester = sender()
     log.debug("Got a request from [{}] to start entity [{}] in shard [{}]", requester, start.entityId, shardId)
-    if (passivateIdleTask.isDefined) {
-      lastMessageTimestamp = lastMessageTimestamp.updated(start.entityId, System.nanoTime())
+    touchLastMessageTimestamp(start.entityId)
+
+    if (state.entities(start.entityId)) {
+      getOrCreateEntity(start.entityId)
+      requester ! ShardRegion.StartEntityAck(start.entityId, shardId)
+    } else {
+      processChange(EntityStarted(start.entityId)) { evt =>
+        getOrCreateEntity(start.entityId)
+        sendMsgBuffer(evt)
+        requester ! ShardRegion.StartEntityAck(start.entityId, shardId)
+      }
     }
-    getOrCreateEntity(
-      start.entityId,
-      _ =>
-        processChange(EntityStarted(start.entityId))(_ =>
-          requester ! ShardRegion.StartEntityAck(start.entityId, shardId)))
```

Comment:
Started entity first and then updated state, which was inconsistent with how it's done for normal messages.
This was introduced in https://github.com/akka/akka/commit/e1a0a1fa3fec7650eb0f28027920357f773e3629